server/server_test.go

```diff
@@ -33,15 +36,53 @@ func TestListenAndServe(t *testing.T) {
 	}
```

Comment:
Let's also check that `td.handler != nil`, since we depend on that in the new test.